wasm-rpc/src/lib.rs

```diff
@@ -304,6 +402,576 @@ fn build_tree(node: &WitNode, nodes: &[WitNode]) -> Value {
+            TypeAnnotatedValue::S64(_) => AnalysedType(golem_wasm_ast::analysis::AnalysedType::S64),
+            TypeAnnotatedValue::U64(_) => AnalysedType(golem_wasm_ast::analysis::AnalysedType::U64),
+            TypeAnnotatedValue::F32(_) => AnalysedType(golem_wasm_ast::analysis::AnalysedType::F32),
+            TypeAnnotatedValue::F64(_) => AnalysedType(golem_wasm_ast::analysis::AnalysedType::F64),
+            TypeAnnotatedValue::Chr(_) => AnalysedType(golem_wasm_ast::analysis::AnalysedType::Chr),
+            TypeAnnotatedValue::Str(_) => AnalysedType(golem_wasm_ast::analysis::AnalysedType::Str),
+            TypeAnnotatedValue::List(value) => value.clone().typ,
+            TypeAnnotatedValue::Tuple(value) => {
+                AnalysedType(golem_wasm_ast::analysis::AnalysedType::Tuple(
+                    value.clone().typ.into_iter().map(|t| t.0).collect(),
+                ))
+            }
+            TypeAnnotatedValue::Record(value) => {
+                AnalysedType(golem_wasm_ast::analysis::AnalysedType::Record(
+                    value
+                        .clone()
+                        .typ
+                        .into_iter()
+                        .map(|(name, ty)| (name, ty.0))
+                        .collect(),
+                ))
+            }
+            TypeAnnotatedValue::Flags(value) => AnalysedType(
+                golem_wasm_ast::analysis::AnalysedType::Flags(value.clone().typ),
+            ),
+            TypeAnnotatedValue::Enum(value) => AnalysedType(
+                golem_wasm_ast::analysis::AnalysedType::Enum(value.clone().typ),
+            ),
+            TypeAnnotatedValue::Option(value) => value.clone().typ,
+            TypeAnnotatedValue::Result(value) => {
+                AnalysedType(golem_wasm_ast::analysis::AnalysedType::Result {
+                    ok: value.clone().ok.map(|value| Box::new(value.0)),
+                    error: value.clone().error.map(|value| Box::new(value.0)),
+                })
+            }
+            TypeAnnotatedValue::Handle(value) => {
+                AnalysedType(golem_wasm_ast::analysis::AnalysedType::Resource {
+                    id: value.clone().id,
+                    resource_mode: value.clone().resource_mode,
+                })
+            }
+            TypeAnnotatedValue::Variant(value) => {
+                AnalysedType(golem_wasm_ast::analysis::AnalysedType::Variant(
+                    value
+                        .clone()
+                        .typ
+                        .into_iter()
+                        .map(|(name, ty)| (name, ty.map(|t| t.0)))
+                        .collect(),
+                ))
+            }
+        }
+    }
+}
+
+impl From<TypeAnnotatedValue> for WitValue {
+    fn from(value: TypeAnnotatedValue) -> Self {
+        let value: Value = value.into();
+        value.into()
+    }
+}
+
+impl WasmValue for TypeAnnotatedValue {
```

Comment:
These instances must be in the `text` module so they are only compiled if the `text` feature flag is on

Reply:
Okay

Reply:
Done